sql/connect/common/src/main/protobuf/spark/connect/pipelines.proto

```diff
@@ -102,19 +109,27 @@ message PipelineCommand {
     // Name of the dataset this flow writes to. Can be partially or fully qualified.
     optional string target_dataset_name = 3;
 
-    // An unresolved relation that defines the dataset's flow. Empty if the query function
-    // that defines the flow cannot be analyzed at the time of flow definition.
-    optional spark.connect.Relation relation = 4;
-
     // SQL configurations set when running this flow.
-    map<string, string> sql_conf = 5;
+    map<string, string> sql_conf = 4;
 
     // Identifier for the client making the request. The server uses this to determine what flow
     // evaluation request stream to dispatch evaluation requests to for this flow.
-    optional string client_id = 6;
+    optional string client_id = 5;
 
     // The location in source code that this flow was defined.
-    optional SourceCodeLocation source_code_location = 7;
+    optional SourceCodeLocation source_code_location = 6;
+
+    oneof details {
+      WriteRelationFlowDetails relation_flow_details = 7;
```

Comment:
open to discussion, wdyt we define a `ImplicitFlowDetails`  that only has `relation`, and `StandaloneFlowDetails` that has `flow_name`, `relation` and potential `optional bool once` in the future.